storage-access.bs

```diff
@@ -162,7 +162,15 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 
     ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). In practice, this might involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
 
-1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global|'s [=environment/has storage access=].
+1. Run the following steps [=in parallel=]:
+    1. Let <dfn>whether the user agent allows unpartitioned cookie access</dfn> be an algorithm that, given a user agent's settings object |user agent settings| and a [=boolean=] |has per-frame storage access|, runs the following steps:
+        1. [=Queue a global task=] on the [=permission task source=] given |global| to:
+            1. If |has per-frame storage access| is false:
+                1. If |user agent settings| allows unpartitioned cookie access, [=/resolve=] |p| with true and return |p|.
+            1. If |user agent settings| disallows unpartitioned cookie access, [=/resolve=] |p| with false and return |p|.
+            1. [=/Resolve=] |p| with true and return |p|.
+    1. Let |user agent settings| be user agent's settings.
+    1. Run <a href="#whether-the-user-agent-allows-unpartitioned-cookie-access">whether the user agent allows unpartitioned cookie access</a> with |user agent settings| and |global|'s [=environment/has storage access=].
```

Comment:
You need to grab the state from the global inside the task steps, not inside the in parallel steps.

And instead of sending all the user agent settings over, why not check for the particular bit you care about while in parallel and send the result of that over?